File: Excel/700-799/735/735 Challenge.py
import pandas as pd

path = "700-799/735/735 Transpose.xlsx"
input = pd.read_excel(path, usecols="A", skiprows=1, nrows=5)
test = pd.read_excel(path, usecols="C:D", skiprows=1, nrows=6)

input_long = input['String'].str.split(', ', expand=True).stack().str.split(' - ', expand=True)
result = (input_long.groupby(1)[0]
          .unique()
          .apply(', '.join)
          .reset_index()
          .sort_values(1)
          .reset_index(drop=True)
          .rename(columns={1: 'Alphabet', 0: 'Planets'}))

print(result.equals(test))
# Earth missing in the expected output.

File: Excel/700-799/735/735 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/700-799/735/735 Transpose.xlsx"
input = read_excel(path, range = "A2:A6")
test = read_excel(path, range = "C2:D7")

result = input %>%
  separate_longer_delim(String, delim = ", ") %>%
  separate_wider_delim(
    String,
    delim = " - ",
    names = c("Planets", "Alphabet")
  ) %>%
  summarise(
    Planets = paste0(unique(Planets), collapse = ", "),
    .by = Alphabet
  ) %>%
  arrange(Alphabet)

all.equal(result, test, check.attributes = FALSE)
# Earth missed in expected result
